internal/services/datafactory/data_factory_pipeline_resource.go

```diff
@@ -109,6 +110,67 @@ func resourceDataFactoryPipeline() *pluginsdk.Resource {
 			},
 		},
 	}
+
+	if features.FourPointOhBeta() {
+		resource.Schema["parameters"] = &pluginsdk.Schema{
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"name": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+					},
+					"default_value": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+					},
+					"type": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+						Default:  string(datafactory.ParameterTypeString),
+						ValidateFunc: validation.StringInSlice([]string{
+							string(datafactory.ParameterTypeString),
+							string(datafactory.ParameterTypeInt),
+							string(datafactory.ParameterTypeFloat),
+							string(datafactory.ParameterTypeBool),
+							string(datafactory.ParameterTypeArray),
+							string(datafactory.ParameterTypeObject),
+							string(datafactory.ParameterTypeSecureString),
+						}, false),
+					},
+				},
+			},
+		}
+
+		resource.Schema["variables"] = &pluginsdk.Schema{
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"name": {
```

Comment:
Add a validation func here, the `name` should not be empty.

Reply:
added